src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -129,86 +167,502 @@ def __repr__(self):
+
+    if allocation.default_when_disabled:
+        featureallocationdict[FeatureFlagConstants.DEFAULT_WHEN_DISABLED] = allocation.default_when_disabled
+
+    return featureallocationdict
+
+
+def custom_serialize_variants(variants_list):
+    """
+    Helper Function to serialize Variants
+
+    Args:
+        variants_list - List of FeatureVariant objects
+
+    Return:
+        JSON serializable List
+    """
+    featurevariants = []
+    for variant in variants_list:
+        variant_dict = {}
+        variant_dict[FeatureFlagConstants.NAME] = variant.name
+        if variant.configuration_value:
+            variant_dict[FeatureFlagConstants.VARIANT_CONFIGURATION_VALUE] = variant.configuration_value
+
+        if variant.status_override:
+            variant_dict[FeatureFlagConstants.VARIANT_STATUS_OVERRIDE] = variant.status_override
+
+        featurevariants.append(variant_dict)
+    return featurevariants
+
+
 def map_featureflag_to_keyvalue(featureflag):
-    '''
-        Helper Function to convert FeatureFlag object to KeyValue object
+    """
+    Helper Function to convert FeatureFlag object to KeyValue object
 
-        Args:
-            featureflag - FeatureFlag object to be converted
+    Args:
+        featureflag - FeatureFlag object to be converted
 
-        Return:
-            KeyValue object
-    '''
+    Return:
+        KeyValue object
+    """
     try:
         enabled = False
         if featureflag.state in ("on", "conditional"):
             enabled = True
 
-        feature_flag_value = FeatureFlagValue(id_=featureflag.name,
-                                              description=featureflag.description,
-                                              enabled=enabled,
-                                              conditions=featureflag.conditions)
-
-        set_kv = KeyValue(key=featureflag.key,
-                          label=featureflag.label,
-                          value=json.dumps(feature_flag_value,
-                                           default=lambda o: o.__dict__,
-                                           ensure_ascii=False),
-                          content_type=FeatureFlagConstants.FEATURE_FLAG_CONTENT_TYPE,
-                          tags={})
+        feature_flag_value = FeatureFlagValue(
+            id_=featureflag.name,
+            description=featureflag.description,
+            enabled=enabled,
+            conditions=featureflag.conditions,
+            display_name=featureflag.display_name,
+            allocation=featureflag.allocation,
+            variants=featureflag.variants,
+            telemetry=featureflag.telemetry,
+        )
+
+        set_kv = KeyValue(
+            key=featureflag.key,
+            label=featureflag.label,
+            value=json.dumps(
+                feature_flag_value, default=lambda o: {k: v for k, v in o.__dict__.items() if v is not None}, ensure_ascii=False
```

Comment:
Why did we change the lambda here?

Reply:
My thinking was not to add new properties to a flag that doesn't have the new properties. If we set them to None the feature flag value would have these properties set to null in its value e.g {id : "test", description" : "this is a description", ... allocation: null, telemetry: null } so we added the lambda to remove the null values in the feature flag object.

Reply:
Does that mean we always add default value for conditions since we initialize it to `{client_filters: []}` ? Even when feature flag contains only variants?

Reply:
I think we should remove the default conditions value while writing new feature flags. While reading, its okay to support reading the default value for backwards compatibility.

Reply:
Yes we will always add default conditions when writing new flags. I am wondering if this would be a breaking change if we remove adding default conditions to the flags, maybe there are users who already expect this behavior.

Reply:
For V1, we should add default conditions. But for V2, we can avoid all defaults. There's no need to have empty conditions defined for a variant. Similarly, no need for empty variants or allocation or telemetry when conditions are specified. While reading, we can be lenient and parse default empty values. But during writing, we should be stricter and not write null or empty values.